File: my-folder/1764-maximum-repeating-substring/solution.java
class Solution {
    public int maxRepeating(String sequence, String word) {
        StringBuilder sb = new StringBuilder();
        int count = 0;
        while (sb.length() <= sequence.length()) {
            sb.append(word);
            if (sequence.contains(sb)) count++;
        }
        return count;
    }
}

File: my-folder/1764-maximum-repeating-substring/solution.py
class Solution:
    def maxRepeating(self, sequence: str, word: str) -> int:
        s = len(sequence)
        w = len(word)

        dp = [0] * (s+1)

        res = 0

        for i in range(w, s+1):
            if (sequence[i-w:i] == word):
                dp[i] = dp[i-w] + 1
                res = max(res, dp[i])
        
        return res
